lib/stdlib/src/json.erl

```diff
@@ -0,0 +1,765 @@
+        noescape, noescape}.
+
+invalid_byte(Bin, Skip) ->
+    Byte = binary:at(Bin, Skip),
+    error({invalid_byte, Byte}, none, error_info(Skip)).
+
+error_info(Skip) ->
+    [{error_info, #{cause => #{position => Skip}}}].
+
+%%
+%% Decoding implementation
+%%
+
+% We use integers instead of atoms for better codegen
+-define(ARRAY, 0).
+-define(OBJECT, 1).
+
+-define(is_1_to_9(X), X =:= $1; X =:= $2; X =:= $3; X =:= $4; X =:= $5; X =:= $6; X =:= $7; X =:= $8; X =:= $9).
+-define(is_0_to_9(X), X =:= $0; ?is_1_to_9(X)).
+-define(is_ws(X), X =:= $\s; X =:= $\t; X =:= $\r; X =:= $\n).
+
+-type from_binary_fun() :: fun((binary()) -> dynamic()).
+-type array_start_fun() :: fun((Acc :: dynamic()) -> ArrayAcc :: dynamic()).
+-type array_push_fun() :: fun((Value :: dynamic(), Acc :: dynamic()) -> NewAcc :: dynamic()).
+-type array_finish_fun() :: fun((ArrayAcc :: dynamic()) -> dynamic()).
+-type object_start_fun() :: fun((Acc :: dynamic()) -> ObjectAcc :: dynamic()).
+-type object_push_fun() :: fun((Key :: dynamic(), Value :: dynamic(), Acc :: dynamic()) -> NewAcc :: dynamic()).
+-type object_finish_fun() :: fun((ObjectAcc :: dynamic()) -> dynamic()).
+
+-type decoders() :: #{
+    empty_array => term(),
+    array_start => array_start_fun(),
+    array_push => array_push_fun(),
+    array_finish => array_finish_fun(),
+    empty_object => term(),
+    object_start => object_start_fun(),
+    object_push => object_push_fun(),
+    object_finish => object_finish_fun(),
+    float => from_binary_fun(),
+    integer => from_binary_fun(),
+    string => from_binary_fun(),
+    null => term()
+}.
+
+-record(decode, {
+    empty_array = [] :: term(),
+    array_start :: array_start_fun() | undefined,
+    array_push :: array_push_fun() | undefined,
+    array_finish = fun lists:reverse/1 :: array_finish_fun(),
+    empty_object = #{} :: term(),
+    object_start :: object_start_fun() | undefined,
+    object_push :: object_push_fun() | undefined,
+    object_finish = fun maps:from_list/1 :: object_finish_fun(),
+    float = fun erlang:binary_to_float/1 :: from_binary_fun(),
+    integer = fun erlang:binary_to_integer/1 :: from_binary_fun(),
+    string :: from_binary_fun() | undefined,
+    null = null :: term()
+}).
+
+-type acc() :: dynamic().
+-type stack() :: [?ARRAY | ?OBJECT | binary() | acc()].
+-type decode() :: #decode{}.
+
+-type simple_decode_value() ::
+    integer() |
+    float() |
+    boolean() |
+    null |
+    binary() |
+    list(simple_decode_value()) |
+    #{binary() => simple_decode_value()}.
+
+-doc "
+Parses a JSON value from `Binary`.
+
+Supports basic data mapping:
+| **JSON** | **Erlang**           |
+|----------|----------------------|
+| Number   | integer() \| float() |
+| Boolean  | true \| false        |
+| Null     | null                 |
+| String   | binary()             |
+| Object   | #{binary() => _}     |
+".
```

Comment:
Does not render a table, [decode doc](https://erlang.github.io/prs/8111/lib/stdlib-5.2/doc/html/json.html#decode/1)